stl/inc/chrono

```diff
@@ -5408,6 +5408,54 @@ _NODISCARD constexpr const _CharT* _Parse_chrono_format_specs(
 }
 
 namespace chrono {
+    // This functionality already exists in put_time, so why is it duplicated here?
+    // Unfortunately, put_time does not like to work with illogical dates, but the standard mandates that these are
+    // printed properly and treated specially. For example, consider operator<< for day. If day is 40, put_time asserts
+    // but the standard requires that it prints "40 is not a valid day".
+    template <class _CharT>
+    bool _Try_simple_write(basic_ostream<_CharT>& _Os, const char _Type, const tm& _Time) {
+        switch (_Type) {
+        case 'd':
+        case 'e':
+            if (_Time.tm_mday < 10) {
+                _Os << (_Type == 'd' ? _CharT{'0'} : _CharT{' '});
+            }
+            _Os << _Time.tm_mday;
+            return true;
+        case 'm':
+            if (_Time.tm_mon + 1 < 10) {
+                _Os << _CharT{'0'};
+            }
+            _Os << _Time.tm_mon + 1;
+            return true;
+        case 'Y':
+            _Os << _STD format(_STATICALLY_WIDEN(_CharT, "{:04}"), _Time.tm_year + 1900);
+            return true;
+        case 'y':
+            _Os << _STD format(_STATICALLY_WIDEN(_CharT, "{:02}"), (_Time.tm_year + 1900) % 100);
+            return true;
+        case 'C':
+            _Os << _STD format(_STATICALLY_WIDEN(_CharT, "{:02}"), (_Time.tm_year + 1900) / 100);
+            return true;

```

Comment:
This doesn't handle negative years correctly. See `year_month_day::_Days_from_civil` or `_Time_parse_fields::_Decompose_year`. Specifically, [[tab:time.format.spec]](http://eel.is/c++draft/tab:time.format.spec) says that `%C` gives "The year divided by 100 using _floored division_", while integer division truncates.

Reply:
Great catch!